From 2f789ca5317725a005114c8e7900a0fb87744272 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Wed, 12 Oct 2016 14:50:26 +0800
Subject: [PATCH 1195/5242] MLK-12928-12 input: keyboard: pf1550: enable ONKEY
 wakeup interrupt before suspend

commit  03cb430a3be74ff03043002392eea7e2f1e8333f from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable ONKEY wakeup interrupt before suspend, and remove usless marocs.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/input/keyboard/pf1550_onkey.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/input/keyboard/pf1550_onkey.c b/drivers/input/keyboard/pf1550_onkey.c
index c1b487a..2ba0d15 100644
--- a/drivers/input/keyboard/pf1550_onkey.c
+++ b/drivers/input/keyboard/pf1550_onkey.c
@@ -142,6 +142,7 @@ static int pf1550_onkey_probe(struct platform_device *pdev)
 	}
 
 	onkey->input = input;
+	onkey->pf1550 = pf1550;
 	platform_set_drvdata(pdev, onkey);
 
 	device_init_wakeup(&pdev->dev, onkey->wakeup);
@@ -159,6 +160,8 @@ static int pf1550_onkey_suspend(struct device *dev)
 			     PF1550_PMIC_REG_ONKEY_INT_MASK0,
 			     ONKEY_IRQ_PUSHI | ONKEY_IRQ_1SI | ONKEY_IRQ_2SI |
 			     ONKEY_IRQ_3SI | ONKEY_IRQ_4SI | ONKEY_IRQ_8SI);
+	else
+		enable_irq_wake(onkey->pf1550->irq);
 
 	return 0;
 }
@@ -173,6 +176,8 @@ static int pf1550_onkey_resume(struct device *dev)
 			     PF1550_PMIC_REG_ONKEY_INT_MASK0,
 			     ~(ONKEY_IRQ_PUSHI | ONKEY_IRQ_1SI | ONKEY_IRQ_2SI |
 			     ONKEY_IRQ_3SI | ONKEY_IRQ_4SI | ONKEY_IRQ_8SI));
+	else
+		disable_irq_wake(onkey->pf1550->irq);
 
 	return 0;
 }
-- 
1.7.9.5

